Document kubectl subresource support (#32604)
kubernetes/website: content/en/docs/reference/kubectl/conventions.md

* Fully-qualify the version. For example, `jobs.v1.batch/myjob`. This will ensure that kubectl does not use its default version that can change over time.
* Don't rely on context, preferences, or other implicit states.

## Subresources

> * You can use the `--subresource` flag for kubectl commands like `get`, `patch`,

Reviewer: Nit: maybe worth mentioning this is alpha flag.

Author: whoops, not sure how I missed that. Fixed :)

> * You can use the `[--subresource=status|scale]` flag for kubectl commands like `get`, `patch`,

Reviewer 2: Would it help to explicitly mention that subresources like `logs` and port forward are not supported similar to how its mentioned in the KEP?

Author: Done.

> `edit` and `replace` to fetch and update `status` and `scale` subresources for all resources that

Reviewer 3: ```suggestion
`edit` and `replace` to fetch and update subresources for all resources
```

> support these subresources. Currently, only these two subresources are supported.

Reviewer 3: ```suggestion
support them. Currently, only the `status` and `scale` subresources are supported.
```

* The API contract against a subresource is identical to a full resource. While updating the

> status subresource to hold a new value, keep in mind that the subresource could be potentially

Reviewer 4: nit
```suggestion
`status` subresource to a new value, keep in mind that the subresource could be potentially
```

Author: @tengqm done.
